testing movement

NotJezus opened the pull request:
sorry forget i changed to inventory branch. but here you have the movementscript testing stuff

NotJezus committed:
Add unit and play mode tests for MovementScript

Introduces edit mode and play mode test scripts for MovementScript, covering ground check setup, component validation, movement, and jumping behavior. Also adds test-only methods to MovementScript for improved testability. Minor project and render pipeline asset updates included.

NotJezus committed:
Remove unused variables from movement tests

Cleaned up MovementScriptPlayModeTests by removing unused variables and redundant code in test methods for improved readability.

NotJezus committed:
Remove unused Test_IsGrounded method

Deleted the Test_IsGrounded method from MovementScript.cs as it was not being used. This helps clean up the code and maintain clarity in the test section.

Ian-kowalski commented on Assets/Scripts/Player/MovementScript.cs:
```diff
@@ -140,4 +139,19 @@ private void UpdateFacingDirection()
             transform.localScale = localScale;
         }
     }
+
+#if UNITY_EDITOR
+    public void Test_SetHorizontal(float value)
+    {
+        horizontal = value;
+        Move();
+    }
+    public void Test_Jump()
+    {
+        if (IsGrounded())
+        {
+            rb.linearVelocity = new Vector2(rb.linearVelocity.x, jumpingPower);
+        }
+    }
+#endif
```
dont like to hve functions just for test but it fine i think

NotJezus replied:
do i need to change it?

Ian-kowalski commented on Assets/Tests/EditMode/MovementScriptEditModeTests.cs:
```diff
@@ -0,0 +1,60 @@
+    {
+        var playerObj = new GameObject("Player");
+        var movement = playerObj.AddComponent<MovementScript>();
+        var groundCheck = new GameObject("GroundCheck").transform;
```
look into having a set up and a break down if ypu have to write the same code often

Ian-kowalski commented on Assets/Tests/PlayMode/MovementScriptPlayModeTests.cs:
```diff
@@ -0,0 +1,97 @@
+    }
+
+    [UnityTest]
+    public IEnumerator Player_Jumps_When_Grounded_And_Jump_Executed()
```
be consistent with name your functions

Ian-kowalski replied:
this is the only one were you have a dash between every word

NotJezus committed:
Refactor jump logic and improve test structure

Moved jump logic into separate jump and jumpAction methods in MovementScript for better modularity and testability. Updated EditMode and PlayMode tests to use more descriptive method names and refactored setup code for clarity and reuse.
